tests/test_dst_components.py

```diff
@@ -0,0 +1,91 @@
+# SPDX-FileCopyrightText: ASSUME Developers
+#
+# SPDX-License-Identifier: AGPL-3.0-or-later
+
+import pyomo.environ as pyo
+import pytest
+
+from assume.units.dst_components import create_electrolyser
+
+
+@pytest.fixture
+def electrolyser_config():
+    return {
+        "rated_power": 50,
+        "min_power": 0,
+        "ramp_up": 30,
+        "ramp_down": 30,
+        "min_operating_time": 1,
+        "min_down_time": 1,
+        "efficiency": 1,
+    }
+
+
+@pytest.fixture
+def electrolyser_model(electrolyser_config):
+    model = pyo.ConcreteModel()
+    time_steps = range(10)
+    model.time_steps = pyo.Set(initialize=time_steps)
+    model.electricity_price = pyo.Param(model.time_steps, initialize=1, mutable=True)
+    model_part = create_electrolyser(
+        model, time_steps=model.time_steps, **electrolyser_config
+    )
+    model.electrolyser = model_part
+    # Objective function to minimize operating cost (Just for testing purpose)
+    model.total_cost = pyo.Objective(
+        expr=sum(
+            model.electrolyser.electrolyser_operating_cost[t] for t in model.time_steps
+        ),
+        sense=pyo.minimize,
+    )
+    # Constraint for total hydrogen production over all time steps (Just for testing purpose)
+    total_hydrogen_production = 300
+    model.total_hydrogen_constraint = pyo.Constraint(
+        expr=sum(model.electrolyser.hydrogen_out[t] for t in model.time_steps)
+        == total_hydrogen_production
+    )
+
+    return model
+
+
+def test_electrolyser_ramping_and_power_bounds(electrolyser_model):
```

Comment:
I find this a very nice way to test the DST components. Maybe also adding some equality checks would make everything even better and more robust? So that some particular values match exactly when everything works as expected? For example checking the total cost to have some specific value at some time steps, or the power being some paticular values. There is no need to check for all of them, just one of the values would be enough